src/llvm/orc/thread_safe_context.cr

```diff
@@ -0,0 +1,30 @@
+{% skip_file if LibLLVM::IS_LT_110 %}
+
+@[Experimental("The C API wrapped by this type is marked as experimental by LLVM.")]
+class LLVM::Orc::ThreadSafeContext
+  protected def initialize(@unwrap : LibLLVM::OrcThreadSafeContextRef)
+  end
+
+  def self.new
+    new(LibLLVM.orc_create_new_thread_safe_context)
+  end
+
+  def to_unsafe
+    @unwrap
+  end
+
+  def dispose : Nil
+    LibLLVM.orc_dispose_thread_safe_context(self)
+    @unwrap = LibLLVM::OrcThreadSafeContextRef.null
+  end
+
+  def finalize
+    if @unwrap
+      dispose
+    end
+  end
+
+  def context : LLVM::Context
+    LLVM::Context.new(LibLLVM.orc_thread_safe_context_get_context(self), false)
```

Comment:
Would it not be better to have `ThreadSafeContext` re-expose all of the `Context` methods (at least builders' creation)? Since the underlying context is set to not `dispose` on `finalize`, all of the builders created by the `Context` will be leaked. Maybe this is fine as very few builders are created, but this sounds like a similar issue to #14861

Reply:
> discussion occurred on discord

synopsis: this should not be the responsibility of `LLVM::Orc::ThreadSafeContext`, fixing #14861 would be the better course of action

this can be marked as resolved*